src/main/java/com/aws/greengrass/ipc/VerifyClientDeviceIdentityOperationHandler.java

```diff
@@ -0,0 +1,113 @@
+import software.amazon.awssdk.aws.greengrass.model.InvalidArgumentsError;
+import software.amazon.awssdk.aws.greengrass.model.ServiceError;
+import software.amazon.awssdk.aws.greengrass.model.UnauthorizedError;
+import software.amazon.awssdk.aws.greengrass.model.VerifyClientDeviceIdentityRequest;
+import software.amazon.awssdk.aws.greengrass.model.VerifyClientDeviceIdentityResponse;
+import software.amazon.awssdk.eventstreamrpc.OperationContinuationHandlerContext;
+import software.amazon.awssdk.eventstreamrpc.model.EventStreamJsonMessage;
+
+import java.util.Optional;
+
+import static com.aws.greengrass.ipc.common.ExceptionUtil.translateExceptions;
+import static software.amazon.awssdk.aws.greengrass.GreengrassCoreIPCService.VERIFY_CLIENT_DEVICE_IDENTITY;
+
+public class VerifyClientDeviceIdentityOperationHandler
+        extends GeneratedAbstractVerifyClientDeviceIdentityOperationHandler {
+    private static final Logger logger = LogManager.getLogger(VerifyClientDeviceIdentityOperationHandler.class);
+    private static final String COMPONENT_NAME = "componentName";
+    private static final String UNAUTHORIZED_ERROR = "Not Authorized";
+    private static final String NO_DEVICE_CREDENTIAL_ERROR = "Client device credential is required";
+    private static final String NO_DEVICE_CERTIFICATE_ERROR = "Client device certificate is required";
+    private final IotAuthClient iotAuthClient;
+    private final String serviceName;
+    private final AuthorizationHandler authorizationHandler;
+
+    /**
+     * Constructor.
+     *
+     * @param context              operation continuation handler
+     * @param iotAuthClient        auth client for client device calls
+     * @param authorizationHandler authorization handler
+     */
+    public VerifyClientDeviceIdentityOperationHandler(
+            OperationContinuationHandlerContext context, IotAuthClient iotAuthClient,
+            AuthorizationHandler authorizationHandler) {
+
+        super(context);
+        this.iotAuthClient = iotAuthClient;
+        serviceName = context.getAuthenticationData().getIdentityLabel();
+        this.authorizationHandler = authorizationHandler;
+    }
+
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.PreserveStackTrace"})
+    @Override
+    public VerifyClientDeviceIdentityResponse handleRequest(VerifyClientDeviceIdentityRequest request) {
+        return translateExceptions(() -> {
+            try {
+                doAuthorizationForClientDevIdentity();
+            } catch (AuthorizationException e) {
+                logger.atWarn().kv("error", e.getMessage()).kv(COMPONENT_NAME, serviceName).log(UNAUTHORIZED_ERROR);
+                throw new UnauthorizedError(e.getMessage());
+            }
+            String certificate = getCertificateFromCredential(request.getCredential());
+            VerifyClientDeviceIdentityResponse response = new VerifyClientDeviceIdentityResponse();
```

Comment:
[nit]

Move this response creation and return inside the `try`. Seems a bit more clear that it will return or throw. ie. the `catch` won't ever return a response except for an exception.

Reply:
Move the response inside the try. Variables should have the minimum possible scope